src/store/channels-list/api.ts

```diff
@@ -4,13 +4,23 @@ import { get, post } from '../../lib/api/rest';
 import { DirectMessage } from './types';
 
 export async function fetchChannels(id: string) {
-  const channels = await get<any>(`/api/networks/${id}/chatChannels`);
-  return await channels.body;
+  try {
+    const channels = await get<any>(`/api/networks/${id}/chatChannels`);
+    return await channels.body;
+  } catch (error: any) {
+    console.log('Error occured while fetching chatChannels ', error?.response?.body?.error); // eg. error.code = ENOTFOUND
+    return [];
```

Comment:
Probably need some design states? Eg. "Authorization required", "Internet not found" ..etc. In order to properly display those errors to the user

Reply:
I think in most cases the user doesn't even know we're polling so it's probably not a huge deal for now. Maybe eventually we need to distinguish between a first time fetch and the background poll.

Reply:
ok got it. i'll record the foreground/background polling in Jira